ui/tooluifactory.cpp

```diff
@@ -50,4 +73,37 @@ bool ToolUiFactory::remotingSupported() const
 
 void ToolUiFactory::initUi()
 {
+
+}
+
+void ToolUiFactory::initUiObject()
+{
+    if (s_ObjectFactoriesInitialized)
+        return;
+    s_ObjectFactoriesInitialized = true;
+
+    PropertyWidget::registerTab<PropertiesTab>(QStringLiteral("properties"),
+                                               ObjectInspectorFactory::tr("Properties"),
+                                               PropertyWidgetTabPriority::First);
+    ObjectBroker::registerClientObjectFactoryCallback<PropertiesExtensionInterface *>(
+        createExtension<PropertiesExtensionClient>);
+    PropertyWidget::registerTab<MethodsTab>(QStringLiteral("methods"),
+                                            ObjectInspectorFactory::tr("Methods"),
+                                            PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<MethodsExtensionInterface *>(
+        createExtension<MethodsExtensionClient>);
+    PropertyWidget::registerTab<ConnectionsTab>(QStringLiteral("connections"),
+                                                ObjectInspectorFactory::tr("Connections"),
+                                                PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<ConnectionsExtensionInterface *>(
+        createExtension<ConnectionsExtensionClient>);
+    PropertyWidget::registerTab<EnumsTab>(QStringLiteral("enums"),
+                                          ObjectInspectorFactory::tr("Enums"),
+                                          PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ClassInfoTab>(QStringLiteral("classInfo"),
+                                              ObjectInspectorFactory::tr("Class Info"),
+                                              PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ApplicationAttributeTab>(
+                QStringLiteral("applicationAttributes"),
+                ObjectInspectorFactory::tr("Attributes"), PropertyWidgetTabPriority::Advanced);
```

Comment:
This looks wrong. What makes the object tool property extensions so special that they need to be in the base class? Ie. why do only these extensions break with a different selection, not those provided by other tools?

Also, why does selection interfere with this at all? initUi() is supposed to be called for all active tools, as soon as they become active. Does this fail, or is that for some reason not enough?